### PR TITLE
Limit signature algorithms used by the bundle endpoint

### PR DESCRIPTION
**Pull request by azdagron**
We need to constrain the ~cipher suites~ signature algorithms available on the bundle endpoint when ACME is enabled since the KeyManager manages the keys and there are limitations on some KeyManagers as to what signature algorithms are
supported (see issue #2302).

This change updates the TLS config to only support ~cipher suites~ signature algorithms that use SHA256 for the ECP256 and RSA2048 keys generated by the autocert package. This aligns to what is supported by AWS KMS.

### CODE REVIEW

**Comment by azdagron** [comment]
@kunzimariano and @marcosy it would be good to have your eyes on this. I don't know if you have the original setup where you can test if this fixes the problem listed in #2302, but if so, I would be in your debt if you gave this a go :)

**Comment by azdagron** [comment]
Something is still off on this. We're investigating.

**Comment by azdagron** [comment]
Turns out cipher suites wasn't the important part, but rather the signature algorithms used during key exchange. @marcosy has been able to verify that this fixes the issue.

File: pkg/server/endpoints/bundle/internal/autocert/autocert.go [hunk 1]
```diff
@@ -1199,6 +1212,17 @@ func (r *lockedMathRand) int63n(max int64) int64 {
 	return n
 }
 
+func supportedSignatureAlgorithms(privKey crypto.Signer) []tls.SignatureScheme {
+	var out []tls.SignatureScheme
+	switch privKey.Public().(type) {
+	case *ecdsa.PublicKey:
```
Comment (rturner3) [inline pkg/server/endpoints/bundle/internal/autocert/autocert.go:1218]: I wonder if we are being too restrictive here.

For example, AWS KMS supports EC P-384 keys and SHA384 (i.e. `tls.ECDSAWithP384AndSHA384`), and it should also be supported for TLS 1.3:
https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28recommended.29

Comment (azdagron) [reply pkg/server/endpoints/bundle/internal/autocert/autocert.go:1218]: Our ACME layer only uses RSA2048 and ECP256 keys...

Comment (rturner3) [reply pkg/server/endpoints/bundle/internal/autocert/autocert.go:1218]: Ah, right. Thanks for reminding me.
https://github.com/spiffe/spire/blob/2632bb31dfd8b8ae5e59f2cd723c78655678093f/pkg/server/endpoints/bundle/acme_auth.go#L120-L123

**Review by rturner3 — APPROVED** [review]

**Pull request merged**